### PR TITLE
chore: Disable implicit tools integration into drawers list

### PR DESCRIPTION
**Pull request by just-boris**
### Description

Implement this proposal: ZuAXAsmOsora

Related links, issue #, if available: n/a

### How has this been tested?

* Updated related tests and added new dev pages for discovered corner-case
* Run screenshot tests in my dev pipeline

<details>
   <summary>Review checklist</summary>

_The following items are to be evaluated by the author(s) and the reviewer(s)._

#### Correctness

- _Changes include appropriate documentation updates._
- _Changes are backward-compatible if not indicated, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#public-apis)._
- _Changes do not include unsupported browser features, see [`CONTRIBUTING.md`](https://github.com/cloudscape-design/components/blob/main/CONTRIBUTING.md#browsers-support)._
- _Changes were manually tested for accessibility, see [accessibility guidelines](https://cloudscape.design/foundation/core-principles/accessibility/)._

#### Security

- _If the code handles URLs: all URLs are validated through [the `checkSafeUrl` function](https://github.com/cloudscape-design/components/blob/main/src/internal/utils/check-safe-url.ts)._

#### Testing

- _Changes are covered with new/existing unit tests?_
- _Changes are covered with new/existing integration tests?_
</details>

---

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: src/app-layout/__tests__/drawers.test.tsx [hunk 1]
```diff
@@ -35,82 +34,14 @@ describeEachAppLayout(size => {
     const { wrapper } = renderComponent(<AppLayout contentType="form" {...emptyDrawerItems} />);
 
     expect(wrapper.findDrawersTriggers()).toHaveLength(0);
-    expect(wrapper.findToolsToggle()).toBeTruthy();
+    expect(wrapper.findToolsToggle()).toBeFalsy();
   });
 
-  test('should apply drawers treatment to the tools if at least one other drawer is provided', () => {
```
Comment (just-boris) [inline src/app-layout/__tests__/drawers.test.tsx:41]: This and other test from this file moved into `runtime-drawers.test.tsx`, because the tools+drawers merging feature is now only available for runtime drawers

File: src/app-layout/utils/use-drawers.ts [hunk 2]
```diff
@@ -95,28 +110,32 @@ export function useDrawers(
...
+
+  function onActiveDrawerChange(newDrawerId: string | undefined) {
     setActiveDrawerId(newDrawerId);
-    fireNonCancelableEvent(ownDrawers?.onChange, newDrawerId);
-  });
+    if (hasOwnDrawers) {
+      fireNonCancelableEvent(ownDrawers?.onChange, newDrawerId);
+    }
+    if (!toolsProps.toolsHide) {
+      fireNonCancelableEvent(toolsProps.onToolsChange, { open: newDrawerId === TOOLS_DRAWER_ID });
```
Comment (just-boris) [inline src/app-layout/utils/use-drawers.ts:124]: Removed repeating `onToolsChange` handlers from other places, this is the only one where we keep it to support compatibility between tools and drawers

File: src/app-layout/visual-refresh/context.tsx [hunk 3]
```diff
@@ -36,7 +36,7 @@ import { useUniqueId } from '../../internal/hooks/use-unique-id';
 
 interface AppLayoutInternals extends AppLayoutProps {
   activeDrawerId: string | undefined;
-  drawers: Array<DrawerItem>;
+  drawers: Array<DrawerItem> | null;
```
Comment (just-boris) [inline src/app-layout/visual-refresh/context.tsx:39]: Empty array `[]` means that `<AppLayout drawers={[]}>` is used and `null` means that there are no drawers registered in any form. This distinction is needed to decide what we render on the right hand side

File: src/app-layout/visual-refresh/layout.tsx [hunk 4]
```diff
@@ -50,7 +50,7 @@ export default function Layout({ children }: LayoutProps) {
 
   // Content gaps on the left and right are used with the minmax function in the CSS grid column definition
   const hasContentGapLeft = isNavigationOpen || navigationHide;
-  const hasContentGapRight = drawersTriggerCount <= 0 || hasOpenDrawer;
+  const hasContentGapRight = drawersTriggerCount === 0 || hasOpenDrawer;
```
Comment (just-boris) [inline src/app-layout/visual-refresh/layout.tsx:53]: I do not think this count can ever be negative number, so decided to simplify for readability

File: src/app-layout/utils/use-drawers.ts (hunk 2 above)
Comment (YueyingLu) [reply src/app-layout/utils/use-drawers.ts:124]: Do we need to check if tools are ignored?

Comment (just-boris) [reply src/app-layout/utils/use-drawers.ts:124]: We still have "multi drawer with tools mode" for runtime API, this is why the condition is here. See `applyToolsDrawer` function how exactly this works

**Review by YueyingLu — APPROVED** [review]
